### ISSUE TITLE
no output

### ISSUE THREAD

**Issue by upengan78** [richardforth/apache2buddy#365]
Hello, thank you for the script.
I am trying to run this 
curl -sL https://raw.githubusercontent.com/richardforth/apache2buddy/master/apache2buddy.pl | perl
as root on Ubuntu 20.04 but the script does not output anything for some time. Is this normal?
-h flag worked. using -v did not show anything more. 

```
perl -v
This is perl 5, version 30, subversion 0 (v5.30.0) built for x86_64-linux-gnu-thread-multi

Distributor ID: Ubuntu
Description:    Ubuntu 20.04 LTS
Release:        20.04
Codename:       focal
```

What can be done? Thank you.
UG

**Comment by richardforth** [comment]
Hi.

It was tested on Ubuntu 20.04 so should be producing output.

The usual hang points are usually further down when parsing log files.

What happens if you just leave it running?

**Comment by upengan78** [comment]
Thank you. sorry for delay in response.

It appears to be stuck at following process - 
```
ps -ef | grep curl

root      225554  225548  0 12:07 pts/1    00:00:00 /usr/bin/curl -s myip.dnsomatic.com
```
Manually running that

```
# curl -vvv -s myip.dnsomatic.com
*   Trying 146.112.255.205:80...
* TCP_NODELAY set
* Connected to myip.dnsomatic.com (146.112.255.205) port 80 (#0)
> GET / HTTP/1.1
> Host: myip.dnsomatic.com
> User-Agent: curl/7.68.0
> Accept: */*
>
^C  
```
Had to CTRL - C to get out of this manual curl command. Not sure why it's not returning anything from there in spite of GET

**Comment by richardforth** [comment]
Sounds like it's getting hung up looking for the up address using curl.

**Comment by richardforth** [comment]
Maybe the site http://myip.dnsomatic.com/ is having issues.

**Comment by richardforth** [comment]
I'm not in a position to test or make changes but I may need to change the
provider URL for gathering the IP address or add a new option to allow
skipping IP address lookups.

**Comment by upengan78** [comment]
Thank you Richard. I ran the script locally and changed that dnsomatic.com to ifconfig.co temporarily and it then ran quickly as expected. Initially i thought myup.dnsomatic.com was having issues but later I checked other systems from our university network and running curl provided the wan-ip as expected. Not sure why the system I was running on was not getting responses to API calls. 

So this worked for me on the problem host: curl -s ifconfig.co 

No problem. Happy holidays. Thank you for your work on this, it's very much appreciated.
